### PR TITLE
MAKE-972: add parallel s3 sync ops

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: operations/sthree.go [hunk 1]
```diff
@@ -267,6 +268,12 @@ func s3SyncToCmd() cli.Command {
 			if err != nil {
 				return errors.Wrap(err, "problem getting new bucket")
 			}
+			syncOpts := pail.ParallelBucketOptions{
+				Workers:      runtime.NumCPU(),
```
Comment (tychoish) [inline operations/sthree.go]: can we add a n argument to make this configurable? we probably want 2x this in most cases?

File: operations/sthree.go [hunk 2]
```diff
@@ -375,6 +395,26 @@ func s3syncFlags(args ...cli.Flag) []cli.Flag {
 			Name:  "timeout",
 			Usage: "specify a timeout for operations, defaults to unlimited timeout if not specified",
 		},
+		cli.BoolFlag{
```
Comment (tychoish) [inline operations/sthree.go]: you can do BoolTFlag, also you can have workers default to 0 and then turn on parallel if you specify a non-zero value for workers

Comment (julianedwards) [reply operations/sthree.go]: Workers defaults to 0 now

File: operations/sthree.go [hunk 3]
```diff
@@ -294,11 +304,20 @@ func s3SyncFromCmd() cli.Command {
 				DryRun:                   c.Bool("dry-run"),
 				DeleteOnSync:             c.Bool("delete"),
 				MaxRetries:               defaultMaxRetries,
+				UseSingleFileChecksums:   true,
```
Comment (tychoish) [inline operations/sthree.go]: only add this if in parallel mode?

File: operations/sthree.go [hunk 4]
```diff
@@ -262,11 +262,21 @@ func s3SyncToCmd() cli.Command {
...
```
Comment (tychoish) [inline operations/sthree.go]: only when parallel?

**Pull request merged**